### ISSUE TITLE
Fails to build on latest Hyprland commit

### ISSUE THREAD

**Issue by matt1432** [horriblename/hyprgrass#58]
Since this [PR](https://github.com/hyprwm/Hyprland/pull/3755), hyprgrass fails to build

```
hyprgrass> The Meson build system
hyprgrass> Version: 1.2.1
hyprgrass> Source dir: /build/l1lfc7xh3m34b0r347gkm13zq96pw3d3-source
hyprgrass> Build dir: /build/l1lfc7xh3m34b0r347gkm13zq96pw3d3-source/build
hyprgrass> Build type: native build
hyprgrass> Project name: hyprgrass
hyprgrass> Project version: 0.0.1
hyprgrass> C compiler for the host machine: gcc (gcc 13.2.0 "gcc (GCC) 13.2.0")
hyprgrass> C linker for the host machine: gcc ld.bfd 2.40
hyprgrass> C++ compiler for the host machine: g++ (gcc 13.2.0 "g++ (GCC) 13.2.0")
hyprgrass> C++ linker for the host machine: g++ ld.bfd 2.40
hyprgrass> Host machine cpu family: x86_64
hyprgrass> Host machine cpu: x86_64
hyprgrass> Compiler for C++ supports arguments -std=c++23: YES
hyprgrass> meson.build:8: WARNING: Consider using the built-in option for language standard version instead of using "-std=c++23".
hyprgrass> Found pkg-config: /nix/store/ifr6srqgpvygd5vp14748d109ri31isv-pkg-config-wrapper-0.29.2/bin/pkg-config (0.29.2)
hyprgrass> Run-time dependency hyprland found: YES 0.31.0
hyprgrass> Run-time dependency wftouch found: YES 0.0
hyprgrass> Run-time dependency pixman-1 found: YES 0.42.2
hyprgrass> Run-time dependency libdrm found: YES 2.4.116
hyprgrass> Build targets in project: 2
hyprgrass>
hyprgrass> hyprgrass 0.0.1
hyprgrass>
hyprgrass>   User defined options
hyprgrass>     auto_features: enabled
hyprgrass>     bindir       : /nix/store/wrlr3k4xrrbz276xll1ygz4c7cq3kgk3-hyprgrass-0.3.0/bin
hyprgrass>     buildtype    : plain
hyprgrass>     includedir   : /nix/store/wrlr3k4xrrbz276xll1ygz4c7cq3kgk3-hyprgrass-0.3.0/include
hyprgrass>     infodir      : /nix/store/wrlr3k4xrrbz276xll1ygz4c7cq3kgk3-hyprgrass-0.3.0/share/info
hyprgrass>     libdir       : /nix/store/wrlr3k4xrrbz276xll1ygz4c7cq3kgk3-hyprgrass-0.3.0/lib
hyprgrass>     libexecdir   : /nix/store/wrlr3k4xrrbz276xll1ygz4c7cq3kgk3-hyprgrass-0.3.0/libexec
hyprgrass>     localedir    : /nix/store/wrlr3k4xrrbz276xll1ygz4c7cq3kgk3-hyprgrass-0.3.0/share/locale
hyprgrass>     mandir       : /nix/store/wrlr3k4xrrbz276xll1ygz4c7cq3kgk3-hyprgrass-0.3.0/share/man
hyprgrass>     prefix       : /nix/store/wrlr3k4xrrbz276xll1ygz4c7cq3kgk3-hyprgrass-0.3.0
hyprgrass>     sbindir      : /nix/store/wrlr3k4xrrbz276xll1ygz4c7cq3kgk3-hyprgrass-0.3.0/sbin
hyprgrass>     wrap_mode    : nodownload
hyprgrass>
hyprgrass> Found ninja-1.11.1 at /nix/store/jyk3j4632ldxsikjq0syjlj16i12jfsx-ninja-1.11.1/bin/ninja
hyprgrass> meson: enabled parallel building
hyprgrass> meson: enabled parallel installing
hyprgrass> building
hyprgrass> build flags: -j16
hyprgrass> [1/5] Compiling C++ object src/gestures/libgestures.a.p/Gestures.cpp.o
hyprgrass> [2/5] Linking static target src/gestures/libgestures.a
hyprgrass> [3/5] Compiling C++ object src/libhyprgrass.so.p/main.cpp.o
hyprgrass> FAILED: src/libhyprgrass.so.p/main.cpp.o
hyprgrass> g++ -Isrc/libhyprgrass.so.p -Isrc -I../src -I/nix/store/spmknpsimm569zylik2d476mibjgycdh-wf-touch-git-dev/include -I/nix/store/1ml7l4qxmlpj131947j502k6jxnvc3ni-glm-0.9.9.8/include -I/nix/store/ln5nawph2b93bdmwn6mkm1ssskf83cz2-pixman-0.42.2/include/pixman-1 -I/nix/store/rf863vb01j35ki1jgvhdhjma0vixdajl-libdrm-2.4.116-dev/include -I/nix/store/rf863vb01j35ki1jgvhdhjma0vixdajl-libdrm-2.4.116-dev/include/libdrm -I/nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include -I/nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/protocols -I/nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/wlroots -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++23 -fPIC -DWLR_USE_UNSTABLE -MD -MQ src/libhyprgrass.so.p/main.cpp.o -MF src/libhyprgrass.so.p/main.cpp.o.d -o src/libhyprgrass.so.p/main.cpp.o -c ../src/main.cpp
hyprgrass> In file included from /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/plugins/PluginAPI.hpp:25,
hyprgrass>                  from ../src/globals.hpp:3,
hyprgrass>                  from ../src/GestureManager.hpp:3,
hyprgrass>                  from ../src/main.cpp:1:
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:32:5: error: 'Vector2D' does not name a type
hyprgrass>    32 |     Vector2D topLeft;
hyprgrass>       |     ^~~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:33:5: error: 'Vector2D' does not name a type
hyprgrass>    33 |     Vector2D bottomRight;
hyprgrass>       |     ^~~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp: In member function 'SWindowDecorationExtents SWindowDecorationExtents::operator*(const double&) const':
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:37:41: error: 'topLeft' was not declared in this scope
hyprgrass>    37 |         return SWindowDecorationExtents{topLeft * scale, bottomRight * scale};
hyprgrass>       |                                         ^~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:37:58: error: 'bottomRight' was not declared in this scope
hyprgrass>    37 |         return SWindowDecorationExtents{topLeft * scale, bottomRight * scale};
hyprgrass>       |                                                          ^~~~~~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp: In member function 'SWindowDecorationExtents SWindowDecorationExtents::floor()':
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:41:17: error: 'topLeft' was not declared in this scope
hyprgrass>    41 |         return {topLeft.floor(), bottomRight.floor()};
hyprgrass>       |                 ^~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:41:34: error: 'bottomRight' was not declared in this scope
hyprgrass>    41 |         return {topLeft.floor(), bottomRight.floor()};
hyprgrass>       |                                  ^~~~~~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:41:53: error: could not convert '{<expression error>, <expression error>}' from '<brace-enclosed initializer list>' to 'SWindowDecorationExtents'
hyprgrass>    41 |         return {topLeft.floor(), bottomRight.floor()};
hyprgrass>       |                                                     ^
hyprgrass>       |                                                     |
hyprgrass>       |                                                     <brace-enclosed initializer list>
hyprgrass> [4/5] Compiling C++ object src/libhyprgrass.so.p/GestureManager.cpp.o
hyprgrass> FAILED: src/libhyprgrass.so.p/GestureManager.cpp.o
hyprgrass> g++ -Isrc/libhyprgrass.so.p -Isrc -I../src -I/nix/store/spmknpsimm569zylik2d476mibjgycdh-wf-touch-git-dev/include -I/nix/store/1ml7l4qxmlpj131947j502k6jxnvc3ni-glm-0.9.9.8/include -I/nix/store/ln5nawph2b93bdmwn6mkm1ssskf83cz2-pixman-0.42.2/include/pixman-1 -I/nix/store/rf863vb01j35ki1jgvhdhjma0vixdajl-libdrm-2.4.116-dev/include -I/nix/store/rf863vb01j35ki1jgvhdhjma0vixdajl-libdrm-2.4.116-dev/include/libdrm -I/nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include -I/nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/protocols -I/nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/wlroots -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -std=c++23 -fPIC -DWLR_USE_UNSTABLE -MD -MQ src/libhyprgrass.so.p/GestureManager.cpp.o -MF src/libhyprgrass.so.p/GestureManager.cpp.o.d -o src/libhyprgrass.so.p/GestureManager.cpp.o -c ../src/GestureManager.cpp
hyprgrass> In file included from /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/plugins/PluginAPI.hpp:25,
hyprgrass>                  from ../src/globals.hpp:3,
hyprgrass>                  from ../src/GestureManager.hpp:3,
hyprgrass>                  from ../src/GestureManager.cpp:1:
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:32:5: error: 'Vector2D' does not name a type
hyprgrass>    32 |     Vector2D topLeft;
hyprgrass>       |     ^~~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:33:5: error: 'Vector2D' does not name a type
hyprgrass>    33 |     Vector2D bottomRight;
hyprgrass>       |     ^~~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp: In member function 'SWindowDecorationExtents SWindowDecorationExtents::operator*(const double&) const':
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:37:41: error: 'topLeft' was not declared in this scope
hyprgrass>    37 |         return SWindowDecorationExtents{topLeft * scale, bottomRight * scale};
hyprgrass>       |                                         ^~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:37:58: error: 'bottomRight' was not declared in this scope
hyprgrass>    37 |         return SWindowDecorationExtents{topLeft * scale, bottomRight * scale};
hyprgrass>       |                                                          ^~~~~~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp: In member function 'SWindowDecorationExtents SWindowDecorationExtents::floor()':
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:41:17: error: 'topLeft' was not declared in this scope
hyprgrass>    41 |         return {topLeft.floor(), bottomRight.floor()};
hyprgrass>       |                 ^~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:41:34: error: 'bottomRight' was not declared in this scope
hyprgrass>    41 |         return {topLeft.floor(), bottomRight.floor()};
hyprgrass>       |                                  ^~~~~~~~~~~
hyprgrass> /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:41:53: error: could not convert '{<expression error>, <expression error>}' from '<brace-enclosed initializer list>' to 'SWindowDecorationExtents'
hyprgrass>    41 |         return {topLeft.floor(), bottomRight.floor()};
hyprgrass>       |                                                     ^
hyprgrass>       |                                                     |
hyprgrass>       |                                                     <brace-enclosed initializer list>
hyprgrass> ninja: build stopped: subcommand failed.
error: builder for '/nix/store/m3xsdwivh6wk11kkmcvh2r3c0428d7jn-hyprgrass-0.3.0.drv' failed with exit code 1;
       last 10 log lines:
       >       |                 ^~~~~~~
       > /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:41:34: error: 'bottomRight' was not declared in this scope
       >    41 |         return {topLeft.floor(), bottomRight.floor()};
       >       |                                  ^~~~~~~~~~~
       > /nix/store/armhmi9vq49w2zhc8bs2x8648l1daqlp-hyprland-0.31.0+date=2023-11-04_55b4f84-dev/include/hyprland/src/SharedDefs.hpp:41:53: error: could not convert '{<expression error>, <expression error>}' from '<brace-enclosed initializer list>' to 'SWindowDecorationExtents'
       >    41 |         return {topLeft.floor(), bottomRight.floor()};
       >       |                                                     ^
       >       |                                                     |
       >       |                                                     <brace-enclosed initializer list>
       > ninja: build stopped: subcommand failed.
```

**Comment by vaxerski** [comment]
should be fixed by https://github.com/hyprwm/Hyprland/commit/32b3d2b4560b4a5c986a564ae2220312c22fcafd

**Comment by matt1432** [comment]
> should be fixed by [hyprwm/Hyprland@32b3d2b](https://github.com/hyprwm/Hyprland/commit/32b3d2b4560b4a5c986a564ae2220312c22fcafd)

Yes that fixed it. Thank you so much for your hard work
